migen/test/test_platform.py

```diff
@@ -0,0 +1,54 @@
+import unittest
+import importlib
+import pkgutil
+import shutil
+
+from migen import *
+from migen.genlib.cdc import MultiReg
+import migen.build.platforms
+
+
+def _make_platform_test_method(mod, name):
+    def platform_test_method(self):
+        plat = importlib.import_module(mod).Platform()
+        m = TestModulePlatform(plat)
+        plat.build(m, run=False, build_name=name, build_dir=name)
+        shutil.rmtree(name)
```

Comment:
Please use [tempfile](https://docs.python.org/3/library/tempfile.html#tempfile.TemporaryDirectory) and temporary directories. Still do the proper cleanup.